src/main/java/org/spongepowered/api/potion/PotionEffectType.java

```diff
@@ -25,10 +25,12 @@
 
 package org.spongepowered.api.potion;
 
+import org.spongepowered.api.attribute.AttributeSource;
+
 /**
  * Represents a possible type of {@link PotionEffect}.
  */
-public interface PotionEffectType {
+public interface PotionEffectType extends AttributeSource {
```

Comment:
It may be useful to add it to Potion, but that may be very confusing as well because it would only be a proxy there. And at the time of the removal of the effect the potion is no longer available. 

Un addition to that think of the possebility to change the potion effects.
The potion itself can have lots of stuff associated to it but they do not effect other potions as well. But if you change the attribute modifiers on one potion you will alter it for all potions of the same type. This is a very bad side effect that the devs will not expect.
Of course it would be cool to have attribute modifiers per potion item that wear down after a certain time but this is not related to MCs Potionsystem is probably out of the scope of this PR.